uips/uip-4.md

```diff
@@ -0,0 +1,102 @@
+    // The randomized validating key for the spend authorization signature.
+    penumbra.crypto.decaf377_rdsa.v1.SpendVerificationKey rk = 4;
+    // NEW: An encryption of the commitment of the input note to the sender's OVK.
+    optional bytes backref_commitment = 7;
+}
+```
+
+Clients MAY populate the `backref_commitment` field with the encrypted note commitment corresponding to the note they are spending. Initially the field will be optional, allowing for a phased adoption period such that clients have time to implement support for `Spend` backreferences.
+
+### Encryption of Spend Backreference
+
+The `backref_commitment` should be encrypted using `ChaCha20-Poly1305`. [RFC-8349](https://datatracker.ietf.org/doc/rfc8439/) specifies that (key, nonce) pairs MUST NOT be reused.
+
+We derive a symmetric key $k$ from the `OutgoingViewingKey` $ovk$ using the BLAKE2b-512 hash function and personalization string `"Penumbra_Backref"`:
+
+```
+k = BLAKE2b-512("Penumbra_Backref", ovk)
+```
+
+One advantage of using a single key is that we can scan all spends using this key without having to do key derivation each time.
+
+A random 16-byte nonce $n$ will be generated and provided as a prefix in the `backref_commitment` field.
+
+Encryption is performed using the $(k, n)$ tuple and outputs the 32-byte ciphertext of the 32-byte note commitment, and a 16-byte MAC. The transmitted data in the `backref_commitment` field consists of a concatenation of the nonce, ciphertext and MAC:
+
+```
+backref_commitment = n || ciphertext || MAC
+```
+
+The `backref_commitment` is thus 64 bytes (16 byte nonce + 32 byte ciphertext + 16 byte MAC).
+
+### EffectHash
+
+Currently the `EffectHash` for the `Spend` action is computed as:
+
+`effect_hash = BLAKE2b-512(len(type_url) || type_url || proto_encode(proto))`
+
+where `type_url` is the bytes of a variable-length Type URL defining the proto message, `len(type_url)` is the length of the Type URL encoded as 8 bytes in little-endian  order, and `proto` represents the proto used to represent the effecting data, and `proto_encode` represents encoding the proto message as a vector of bytes.
+
+#### Backwards Compatibility
+
+The `EffectHash` computation is unchanged if the new `backref_commitment` field is not populated. This is the behavior provided by making the field `optional` in the `SpendBody` protocol message. However, if the `backref_commitment` field is populated, then it is included in the `EffectHash` computation per the existing `proto_encode` method as described above.
+
+### Transaction Perspectives and Views
+
+The `TransactionPerspective` and `TransactionView` will be unchanged. The backreference is treated as an internal sync optimization detail.
+
+## Rationale
+
+ZCash has considered a similar approach wherein backwards syncing is enabled using references in the memo fields. The advantage of this approach is that DAGSync-aware clients can populate these fields without a change to the consensus rules.
```

Comment:
```suggestion
ZCash has considered a similar approach wherein backwards syncing is enabled using references encoded into the memo fields. Wallets can periodically construct transactions that stuff references to previous transaction hashes into the memo field of the dummy transaction. The advantage of the memo-stuffing approach is that DAGSync-aware clients can populate these fields without a change to the consensus rules. The disadvantage, however, is that the user's transaction history is polluted with dummy transactions, and a client must scan forward to find one of these dummy transactions before it can go backwards.
```